src/hooks/useDebounce.js

```diff
@@ -0,0 +1,31 @@
+import { useCallback } from 'react';
+
+export const useDebounce = (func, waitAmount, immediate) => {
+  const debounceIt = () => {
+    // Closure timeout ref for all subsequent calls
+    let timeout;
+
+    // Return a func wrapper with the timeout var ref closure
+    return (...args) => {
+      const doItLater = () => {
+        timeout = null;
+        if (!immediate) func(...args);
+        return null;
```

Comment:
Prob can just have no `return` here, maybe `undefined` makes more sense in this context than `null`?